### PR TITLE
Only warn about auto-regeneration bug on Bash On Windows.

### LINKED ISSUE #5462: Autoregeration disabled on Windows
I am on (or have tested on) **_Windows**_ 10+
jekyll  version 3.3,

```
D:\Jekyll>jekyll serve
WARN: Unresolved specs during Gem::Specification.reset:
      jekyll-watch (~> 1.1)
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
Configuration file: D:/Jekyll/_config.yml
Configuration file: D:/Jekyll/_config.yml
            Source: D:/Jekyll
       Destination: D:/Jekyll/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
                    done in 0.575 seconds.
                    --watch arg is unsupported on Windows.
                    If you are on Windows Bash, please see: https://github.com/Microsoft/BashOnWindows/issues/216
Configuration file: D:/Jekyll/_config.yml
    Server address: http://127.0.0.1:4000/
  Server running... press ctrl-c to stop.
```

default `Gemfile.lock` file user jekyll-watch 1.5,but jekyll-watch 1.5 listen (~> 3.0, < 3.1)

### PR DESCRIPTION
**Pull request by envygeeks**

### CODE REVIEW

**Comment by ashmaroli** [comment]
Do you mean test it out..? I'll volunteer..

**Comment by parkr** [comment]
> Do you mean test it out..? I'll volunteer..

Yep.

**Comment by ashmaroli** [comment]
Test passed.. was able to view regenerated site successfully..

File: lib/jekyll/commands/build.rb [hunk 1]
```diff
@@ -71,10 +71,10 @@ def build(site, options)
         #
         # Returns nothing.
         def watch(site, options)
-          if Utils::Platforms.windows?
-            Jekyll.logger.warn "", "--watch arg is unsupported on Windows. "
-            Jekyll.logger.warn "", "If you are on Windows Bash, please see: " \
-              "https://github.com/Microsoft/BashOnWindows/issues/216"
+          if Utils::Platforms.bash_on_windows?
+            Jekyll.logger.warn "", "--watch arg is unsupported in Bash on Windows. "
+            Jekyll.logger.warn "", "Please see: https://github.com/Microsoft/BashOnWindows/issues/216"
```
Comment (parkr) [inline lib/jekyll/commands/build.rb]: Looks like this line is too long per the `fit` build failure.

Comment (ashmaroli) [inline lib/jekyll/commands/build.rb]: *`fmt` :point_up_2:

Comment (parkr) [inline lib/jekyll/commands/build.rb]: Yep. I review on my phone (with autocorrect) a lot.

File: lib/jekyll/commands/build.rb [hunk 2]
```diff
@@ -71,10 +71,10 @@ def build(site, options)
...
+            Jekyll.logger.warn "", "If iNotify is fixed, please file a ticket."
```
Comment (Daniel15) [inline lib/jekyll/commands/build.rb]: `inotify` (all lowercase). It's not an Apple product 😛

**Comment by Daniel15** [comment]
Thanks! This fixes watching for me on Windows.

File: lib/jekyll/commands/build.rb (hunk 2 above)
Comment (envygeeks) [inline lib/jekyll/commands/build.rb]: I'll care about that when people start saying GNU/Linux instead of Linux. ;)

**Comment by parkr** [comment]
Fixes #5462.

**Comment by envygeeks** [comment]
I'll fix the fmt error in the morning.

**Comment by envygeeks** [comment]
oh wait no, I was supposed to remove this entirely, BashOnWindows will support iNotify in the next release.

**Comment by envygeeks** [comment]
I'll adjust the pull to reflect that in the morning.

**Comment by parkr** [comment]
> oh wait no, I was supposed to remove this entirely, BashOnWindows will support iNotify in the next release.

Oh nice!

**Comment by parkr** [comment]
Hey @envygeeks, what else needs to be done here?

**Comment by envygeeks** [comment]
I think I need to fix the format error, I'll get it done in the morning, I've ben so busy I forgot about this.

**Comment by parkr** [comment]
Moving from disabling watch altogether to just warning in case they have issues.

**Comment by parkr** [comment]
@jekyllbot: merge +bug

**Pull request merged**